runners/twister2/src/main/java/org/apache/beam/runners/twister2/Twister2LegacyRunner.java

```diff
@@ -0,0 +1,339 @@
+    System.setProperty("job_file", options.getJobFileZip());
+    System.setProperty("job_type", options.getJobType());
+    // do a simple config dir validation
+    System.setProperty("config_dir", options.getTwister2Home() + "/conf/");
+    File cDir = new File(System.getProperty("config_dir"), options.getClusterType());
+
+    String[] filesList =
+        new String[] {
+          "core.yaml", "network.yaml", "data.yaml", "resource.yaml", "task.yaml",
+        };
+
+    for (String file : filesList) {
+      File toCheck = new File(cDir, file);
+      if (!toCheck.exists()) {
+        throw new Twister2RuntimeException(
+            "Couldn't find " + file + " in config directory specified.");
+      }
+    }
+
+    // setup logging
+    FileInputStream fis = null;
+    try {
+      fis = new FileInputStream(new File(cDir, "logger.properties"));
+      LogManager.getLogManager().readConfiguration(fis);
+      fis.close();
+    } catch (IOException e) {
+      LOG.warning("Couldn't load logging configuration");
+    } finally {
+      if (fis != null) {
+        try {
+          fis.close();
+        } catch (IOException e) {
+          LOG.info(e.getMessage());
+        }
+      }
+    }
+  }
+
+  private void setupSystemTest(Twister2PipelineOptions options) {
+    prepareFilesToStage(options);
+    zipFilesToStage(options);
+    System.setProperty("job_file", options.getJobFileZip());
+    System.setProperty("job_type", options.getJobType());
+  }
+
+  private Set<String> extractNames(Set<TSet> leaves) {
+    Set<String> results = new HashSet<>();
+    for (TSet leaf : leaves) {
+      results.add(leaf.getId());
+    }
+    return results;
+  }
+
+  private Map<String, String> extractNames(Map<String, BatchTSet<?>> sideInputs) {
+    Map<String, String> results = new LinkedHashMap<>();
+    for (Map.Entry<String, BatchTSet<?>> entry : sideInputs.entrySet()) {
+      results.put(entry.getKey(), entry.getValue().getId());
+    }
+    return results;
+  }
+
+  /**
+   * Classpath contains non jar files (eg. directories with .class files or empty directories) will
+   * cause exception in running log.
+   */
+  private void prepareFilesToStage(Twister2PipelineOptions options) {
+    List<String> filesToStage =
+        options.getFilesToStage().stream()
+            .map(File::new)
+            .filter(File::exists)
+            .map(
+                file -> {
+                  return file.getAbsolutePath();
+                })
+            .collect(Collectors.toList());
+    options.setFilesToStage(
+        PipelineResources.prepareFilesForStaging(
+            filesToStage,
+            MoreObjects.firstNonNull(
+                options.getTempLocation(), System.getProperty("java.io.tmpdir"))));
+  }
+
+  /**
+   * creates a single zip file from all the jar files that are listed as files to stage in options.
+   *
+   * @param options
+   */
+  private void zipFilesToStage(Twister2PipelineOptions options) {
```

Comment:
@RyanSkraba Thank you for reviewing the code, It is good to get another set of eyes to notice mistakes that one makes. 

A single zip file is being created for each test since Twister2 treats each test as a new job, the zip file contains the dependency files that are needed by the run. I will look into a way to delete the zip files created by tests that have completed so it does not fill up the disk space.

Reply:
I found a solution with forkEvery, I set the temp files created to delete on exit and since the JVM changes for every test class the old temp files will get deleted automatically.